src/conversations/index.js

```diff
@@ -0,0 +1,120 @@
+const cfg = require('../config');
+const client = require('twilio')(cfg.twilioAccountSid, cfg.twilioAuthToken);
+const conversations = require('./conversations');
+const { Conversation, PhoneNumber } = require('../db');
+
+let conversation;
+
+/**
+ * @param {string} baseUrl
+ * @return {Promise<Conversation>}
+ */
+async function getConversation(baseUrl) {
+  if (!conversation) {
+    conversation = await Conversation.get();
+    let twilioConversation;
+    if (conversation) {
+      twilioConversation = await conversations.getConversation(
+        client,
+        conversation.sid
+      );
+    }
+
+    if (!twilioConversation) {
+      twilioConversation = await conversations.createConversation(
+        client,
+        '2 Way Masked SMS'
+      );
+      conversation = await new Conversation({
+        sid: twilioConversation.sid,
+      }).save();
+    }
+    const webhooks = await conversations.getConversationWebhooks(
+      client,
+      conversation.sid
+    );
+    await conversations.updateConversationWebhooks(
+      client,
+      conversation.sid,
+      webhooks,
+      baseUrl
+    );
+  }
+  return conversation;
+}
+
+/**
+ * Add a participant to the conversation and the phone number to the DB
+ * @param {Conversation} conversation
+ * @param {string} number
+ * @param {number} expirationTime
+ * @return {Promise<null|PhoneNumber>}
+ */
+async function addPhoneNumber(conversation, number, expirationTime) {
+  // add a participant
+  const phoneNumber = await PhoneNumber.getByNumber(number);
+  if (!phoneNumber) {
+    const participant = await conversations.addParticipant(
+      client,
+      conversation.sid,
+      number
+    );
+    return await new PhoneNumber({
```

Comment:
the `await` here is unnecessary. It anyways returns a Promise in either way

Reply:
Applies throughout

Reply:
👍